tests/kernel/install_ltp.pm

```diff
@@ -210,14 +211,39 @@ sub install_from_git {
     assert_script_run 'make -j$(getconf _NPROCESSORS_ONLN)', timeout => $timeout;
     script_run 'export CREATE_ENTRIES=1';
     assert_script_run 'make install', timeout => 360;
-    assert_script_run "find /opt/ltp/ -name '*.run-test' > ~/openposix-test-list-$tag";
+    assert_script_run "find /opt/ltp -name '*.run-test' > ~/openposix-test-list-$tag";
 }
 
 sub install_from_repo {
-    my ($tag) = @_;
-    zypper_call('in qa_test_ltp', dumb_term => 1);
-    script_run 'rpm -q qa_test_ltp | tee /opt/ltp_version';
-    assert_script_run q(find ${LTPROOT:-/opt/ltp}/testcases/bin/openposix/conformance/interfaces/ -name '*.run-test' > ~/openposix-test-list-) . $tag;
+    my ($tag)  = @_;
+    my $is_sle = is_sle;
+    my $is_qam = $is_sle && is_released;
+    my $stable = get_var('LTP_STABLE', $is_qam);
+    my $pkg    = get_var('LTP_PKG', ($stable && $is_sle) ? 'qa_test_ltp' : 'ltp');
+    my $repo   = get_var('LTP_REPOSITORY');
+
+    if (!$repo) {
+        if ($is_sle) {
+            add_qa_head_repo;
+            return;
+        }
+
+        my $arch = '';
```

Comment:
`my $arch;` can be used

Reply:
Again, I suspect `Use of uninitialized value in concatenation (.) or string at ...` for intel.

Reply:
Ok, but it just demonstrates how duct taping of if-else spaghetti auto generating repository monster is not correct approach :). If you make sure that variables are correctly set in medium you can drop almost 20 lines of not necessary code. But If you like this approach, I don't insist on change at the moment.